content/blog/2018-05-09-onboarding-social-weather.md

```diff
@@ -0,0 +1,672 @@
+
+``` r
+library("igraph")
+library("ggraph")
+
+set.seed(2016)
+
+word_cors %>%
+  dplyr::filter(correlation > .35) %>%
+  graph_from_data_frame() %>%
+  ggraph(layout = "fr") +
+  geom_edge_link(aes(edge_alpha = correlation), show.legend = FALSE) +
+  geom_node_point(color = "lightblue", size = 5) +
+  geom_node_text(aes(label = name), repel = TRUE) +
+  theme_void()
+```
+
+![](/img/blog-images/2018-05-09-onboarding-social-weather/unnamed-chunk-14-1.png)
+
+This figure gives a good sample of things discussed in reviews. Despite
+our efforts filtering words specific to issues, some of them remain very
+specific, such as country/city/location that are very frequent in
+`ropenaq` review.
+
+#### How positive is onboarding?
+
+Using sentiment analysis, we can look how positive lines are.
+
+``` r
+sentiments %>%
+  dplyr::group_by(role) %>%
+  skimr::skim(sentiment)
+```
+
+    ## Skim summary statistics
+    ##  n obs: 11553 
+    ##  n variables: 6 
+    ##  group variables: role 
+    ## 
+    ## Variable type: numeric 
+    ##               role  variable missing complete    n  mean   sd   min p25
+    ##             author sentiment       0     4823 4823 0.07  0.21 -1.2    0
+    ##  community_manager sentiment       0       97   97 0.13  0.21 -0.41   0
+    ##             editor sentiment       0     1521 1521 0.13  0.22 -1.63   0
+    ##              other sentiment       0      344  344 0.073 0.2  -0.6    0
+    ##           reviewer sentiment       0     4768 4768 0.073 0.21 -1      0
+    ##  median  p75  max     hist
+    ##   0     0.17 1.84 <U+2581><U+2581><U+2582><U+2587><U+2581><U+2581><U+2581><U+2581>
+    ##   0.071 0.23 1    <U+2581><U+2581><U+2587><U+2585><U+2582><U+2581><U+2581><U+2581>
+    ##   0.075 0.25 1.13 <U+2581><U+2581><U+2581><U+2581><U+2587><U+2586><U+2581><U+2581>
+    ##   0     0.2  0.81 <U+2581><U+2581><U+2582><U+2587><U+2582><U+2582><U+2581><U+2581>
+    ##   0     0.17 1.73 <U+2581><U+2581><U+2587><U+2585><U+2581><U+2581><U+2581><U+2581>
+
+``` r
+summary(sentiments$sentiment)
+```
+
+    ##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
+    ## -1.63200  0.00000  0.00000  0.07961  0.18353  1.84223
+
+``` r
+sentiments %>%
+  dplyr::filter(!role %in% c("other", "community_manager")) %>%
+  ggplot(aes(role, sentiment)) +
+  geom_boxplot(fill = "salmon") +
+  hrbrthemes::theme_ipsum(base_size = 16,
+                          axis_title_size = 16,
+                          strip_text_size = 16)
+```
+
+![Sentiment of onboarding review threads by
+line](/img/blog-images/2018-05-09-onboarding-social-weather/unnamed-chunk-15-1.png)
+
+Generally lines are quite positive, although it’d be better to be able
+to compare them with text from traditional review processes of
+scientific manuscripts. So we do get negative lines… about what? Note
+that we have a [code of
+conduct](https://github.com/ropensci/onboarding/blob/master/policies.md/#code-of-conduct)
```

Comment:
> Note that we have a code of conduct that’s always been respected, so you wouldn’t find really mean stuff.

I get your point, but having a code doesn't mean we won't get negative stuff, and I don't know of an example in onboarding where the code had to be brought up so maybe better to omit this or rephrase to indicate we have a code of conduct that helps people recognize our intent to keep things collegial (negative does not equal bad).

Reply:
Ah the sentence was to mention that we have one and that it's always been respected, not that it prevented mean stuff to happen. I'll omit since I phrased it so badly :grin: